src/__test__/table-theme-colors.spec.js

```diff
@@ -0,0 +1,481 @@
+import tableThemeColors from '../table-theme-colors';
+
+describe('table-theme-colors', () => {
+  const spyQuerySelector = jest.fn();
+  const spyGetComputedStyle = jest.fn();
+  const spyGetStyle = jest.fn();
+  const defaultValue = {
+    tableBackgroundColorFromTheme: 'inherit',
+    backgroundColor: undefined,
+    isBackgroundTransparentColor: false,
+    isBackgroundDarkColor: false,
+    borderColor: '#D9D9D9',
+    body: { borderColor: '#D9D9D9' },
+    pagination: {
+      borderColor: '#D9D9D9',
+      color: '#404040',
+      iconColor: 'rgba(0, 0, 0, 0.54)',
+      disabledIconColor: 'rgba(0, 0, 0, 0.3)',
+    },
+  };
+  const darkBackgroundColorValue = {
+    tableBackgroundColorFromTheme: 'inherit',
+    backgroundColor: undefined,
+    body: {
+      borderColor: ' #F2F2F2',
+    },
+    borderColor: ' #F2F2F2',
+    isBackgroundDarkColor: true,
+    isBackgroundTransparentColor: false,
+    pagination: {
+      borderColor: ' #F2F2F2',
+      color: 'rgba(255, 255, 255, 0.9)',
+      disabledIconColor: 'rgba(255, 255, 255, 0.3)',
+      iconColor: 'rgba(255, 255, 255, 0.9)',
+    },
+  };
+  let theme;
+
+  beforeEach(() => {
+    global.document.querySelector = spyQuerySelector;
+    global.window.getComputedStyle = spyGetComputedStyle;
+    theme = {
+      getStyle: spyGetStyle,
+    };
+    spyQuerySelector.mockReturnValue(undefined);
+    spyGetComputedStyle.mockReturnValue(undefined);
+    spyGetStyle.mockReturnValue(undefined);
+  });
+
+  describe('when there is no background color from theme or css file', () => {
+    it('should return the default value', () => {
+      const result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
+    });
+  });
+
+  describe('when there is only background color on sheet', () => {
+    it('should return the default value ', () => {
+      global.document.querySelector = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') return selector;
+        return undefined;
+      });
+      global.window.getComputedStyle = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') {
+          return { backgroundColor: '#fff' };
+        }
+        return undefined;
+      });
+      let result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
+
+      global.window.getComputedStyle = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') {
+          return { backgroundColor: '#000' };
+        }
+        return undefined;
+      });
+      result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
```

Comment:
Is this what we want in this case? the sheet background is dark, but `isBackgroundDarkColor` is false?

Reply:
good question. 
I realized that in client, objectBackgroundColorFromCSS will at least be rgba(0, 0, 0, 0) which is impossible to be undefine. 
I cannot figure out why.